### ISSUE TITLE
flaky test: TestCheckpoint(actuated-arm64-6cpu-8gb, 1.21.x)

### ISSUE THREAD

**Issue by lifubang** [opencontainers/runc#4274]
Sometimes fail:
```
=== RUN   TestCheckpoint
    checkpoint_test.go:115: === /tmp/TestCheckpoint1387080808/003/criu-parent/dump.log ===
    checkpoint_test.go:115: open /tmp/TestCheckpoint1387080808/003/criu-parent/dump.log: no such file or directory
    checkpoint_test.go:119: CRIU feature check failed
--- FAIL: TestCheckpoint (0.24s)
```

Sometimes success to skip:
```
=== RUN   TestCheckpoint
    checkpoint_test.go:115: === /tmp/TestCheckpoint194762558/003/criu-parent/dump.log ===
    checkpoint_test.go:115: open /tmp/TestCheckpoint194762558/003/criu-parent/dump.log: no such file or directory
    checkpoint_test.go:117: criu is missing features: [MemTrack]
--- SKIP: TestCheckpoint (0.51s)
```

**Comment by kolyshkin** [comment]
I am working on that. The issue is, criu feature check is not doing what it should.

**Comment by kolyshkin** [comment]
Interesting; I've never seen it being able to skip! I guess it was skipping when using a runner with an older kernel.

I found two reasons for the test to fail.

1. actuated-arm64 currently runs a kernel with `CONFIG_NF_TABLES_INET` not set, which uncovers a bug in criu (see https://github.com/checkpoint-restore/criu/pull/2403).

2. It seems that the feature check test is not working like it should. I've added a patch to not test pre-dump if memory tracking is not available (one of commits in PR #4276).

These two PRs address the issue.
